File: Api/Models/Class.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace ArknightsApp.Models;

public class Class
{
    public int Id { get; set; }

    [Required]
    [MaxLength(50)]
    public string Name { get; set; } = string.Empty;

    [MaxLength(200)]
    public string Description { get; set; } = string.Empty;

    // Связи
    public ICollection<Subclass> Subclasses { get; set; } = new List<Subclass>();
    public ICollection<Operator> Operators { get; set; } = new List<Operator>();
}

File: client/src/types/class.ts
export interface Class {
    id: number;
    name: string;
    description: string;
}

export interface Subclass {
    id: number;
    name: string;
    description: string;
    classId: number;
}

export interface CreateSubclass {
    name: string;
    description: string;
    classId: number;
}